src/sentry/issues/escalating.py

```diff
@@ -154,36 +167,167 @@ def _query_with_pagination(
 ) -> List[GroupsCountResponse]:
     """Query Snuba for event counts for the given list of project ids and groups ids in
     a time range."""
+    organization = Organization.objects.get(id=organization_id)
     all_results = []
     offset = 0
+
     while True:
-        query = _generate_query(project_ids, group_ids, offset, start_date, end_date, category)
+        query = _generate_entity_dataset_query(
+            project_ids, group_ids, offset, start_date, end_date, category
+        )
         request = Request(
             dataset=_issue_category_dataset(category),
             app_id=REFERRER,
             query=query,
             tenant_ids={"referrer": REFERRER, "organization_id": organization_id},
         )
         results = raw_snql_query(request, referrer=REFERRER)["data"]
+
         all_results += results
         offset += ELEMENTS_PER_SNUBA_PAGE
         if not results or len(results) < ELEMENTS_PER_SNUBA_PAGE:
             break
 
+    if category == GroupCategory.ERROR and features.has(
+        "organizations:escalating-issues-v2", organization
+    ):
+        metrics_results = []
+        metrics_offset = 0
+        limit = math.floor(ELEMENTS_PER_SNUBA_PAGE / (7 * 24)) or 1
+
+        while True:
+            # Generate and execute the query to the Generics Metrics Backend
+            metrics_query = _generate_generic_metrics_backend_query(
+                organization_id,
+                project_ids,
+                group_ids,
+                start_date,
+                end_date,
+                metrics_offset,
+                limit,
+                category,
+            )
+            projects = Project.objects.filter(id__in=project_ids)
+            metrics_series_results = get_series(
+                projects=projects,
+                metrics_query=metrics_query,
+                use_case_id=UseCaseID.ESCALATING_ISSUES,
+            )
+
+            metrics_results += transform_to_groups_count_response(metrics_series_results)
+            metrics_offset += limit
+
+            if (
+                not metrics_series_results["groups"]
+                or len(metrics_series_results["groups"]) < limit
+            ):
+                break
+
+        # Log exception if results from the Metrics backend are
+        # not equivalent to the Errors dataset.
+        if not compare_lists(metrics_results, all_results):
+            logger.info(
+                "Generics Metrics Backend query results not the same as Errors dataset query.",
+                extra={"metrics_results": metrics_results, "dataset_results": all_results},
```

Comment:
Would these being very large be a problem for logging? We _may_ (I'm not sure how we're billed) have to pay a lot more if this line needs to be in place long-term.